### PR TITLE
[BEAM-1340] Move assert_that, equal_to, is_empty to apache_beam.testing.util

### PR DESCRIPTION
**Pull request by charlesccychen**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by charlesccychen** [comment]
R: @robertwb 
CC: @aaltay, @markflyhigh

**Comment by aaltay** [comment]
LGTM

File: sdks/python/apache_beam/transforms/util.py [hunk 1]
```diff
@@ -38,9 +38,6 @@
     'KvSwap',
     'RemoveDuplicates',
     'Values',
-    'assert_that',
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/util.py]: Is it intentional that the implementations were not removed as well?

**Comment by robertwb** [comment]
Can you also rebase? Looks like there's conflicts.

**Comment by charlesccychen** [comment]
Thanks, PTAL.

File: sdks/python/apache_beam/transforms/util.py (hunk 1 above)
Comment (charlesccychen) [inline sdks/python/apache_beam/transforms/util.py]: Thanks, that was an oversight.

**Comment by robertwb** [comment]
LGTM assuming tests pass.

**Comment by charlesccychen** [comment]
Tests at the latest de36138986fb1f858604c80a950443a501f201dd pass locally.

**Comment by robertwb** [comment]
************* Module apache_beam.transforms.util
W: 23, 0: Unused pvalue imported from apache_beam (unused-import)
W: 24, 0: Unused window imported from apache_beam.transforms (unused-import)
W: 26, 0: Unused Create imported from apache_beam.transforms.core (unused-import)
W: 30, 0: Unused WindowInto imported from apache_beam.transforms.core (unused-import)

**Comment by robertwb** [comment]
I can fix these in the merge.

**Comment by charlesccychen** [comment]
Thanks; sorry I missed that--lint is fixed at the latest 5f9f7b6a8bd1bd0ae7313ddd7cd4a86849106820.

**Pull request closed without merging**